### ISSUE TITLE
Canon CR3 Burst-mode support

### ISSUE THREAD

**Issue by vvpeter** [darktable-org/darktable#15096]
DT doesn't seem to support the CR3 RAW burst-mode file.  It only shows the initial image from it.  It would be nice to support all the frames available in the file, as well as the ability to extract them all in 1 operation.

**Comment by Nilvus** [comment]
I don't know that mode but RAW support for CR3 is provided by Libraw. So you probably need to ask to Libraw team on their repository before. I am quite sure darktable team can't do anything about that. @kmilos: as you know more that part, if you have something to add to my comment!

**Comment by cytrinox** [comment]
Not what you requested but maybe it helps:
https://github.com/dnglab/dnglab can extract all frames as DNG with a single command:

    dnglab convert --image-index=all burst_roll.cr3 /tmp/all_frames/

**Comment by vvpeter** [comment]
I tried dnglab but it returns errors when trying to process the file (the analyze option returned: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Unsupported { what: "Unknown camera", model: "Canon EOS R6m2", make: "Canon", mode: "raw" }', bin\dnglab\src\analyze.rs:36:61).  But that would be a handy utility if it supported the file format and/or camera (not real sure which is causing it to fail).

**Comment by MStraeten** [comment]
R6m2 isn’t supported by libraw yet - see https://github.com/darktable-org/darktable/pull/13684

**Comment by vvpeter** [comment]
That's unfortunate 😒.  We'll keep waiting anxiously!

**Comment by kmilos** [comment]
[LibRaw does list raw burst mode support since the last 0.21 version](https://www.libraw.org/download#changelog).

**Comment by vvpeter** [comment]
I downloaded the new version.  The analyze option now works, but when I try the extract option I get this error: Error: General error: TIFF: don't know marker 0x0.

**Comment by rmonat** [comment]
I think this issue still happens in darktable 4.6.1. When opening a CR3 burst in darktable, I just see the first frame.

**Comment by vvpeter** [comment]
Yes, I think the issue still exists, even though libraw v0.21.0 lists support of "Canon CR3 filmrolls/RawBurst".  So I'm not sure if it's now back in the dt court to leverage the functionality and determine how best to handle those files.  I'm assuming that dt would need to extract all the images as separate filesystem files as an initial step so the xmp sidecars could be generated.  Otherwise it would have to be a single xmp that somehow manages all the images contained w/in the single file and I'm not sure if dt supports that technique.

**Comment by Nikita-Presnov** [comment]
Are there any plans to add support for this file type, or at least import of the entire series?

I understand that the main difficulty lies in the simplicity of the GUI

**Comment by MStraeten** [comment]
as long as no one contributes the required code to handle raw burst mode there's no plan.
